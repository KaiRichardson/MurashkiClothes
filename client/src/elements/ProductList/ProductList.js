import styled from 'styled-components';

import { spacing } from 'utils';

export const ProductList = styled.section`
  --gap: ${spacing.lg};

  padding: var(--gap) 0;
  display: grid;
  justify-content: center;
  justify-items: center;
  grid-gap: var(--gap);

  @media screen and (min-width: 768px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media screen and (min-width: 992px) {
    grid-template-columns: repeat(3, 1fr);
  }
`;
