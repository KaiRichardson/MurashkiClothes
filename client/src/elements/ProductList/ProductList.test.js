import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { ProductList } from './ProductList';

describe('<ProductList /> tests', () => {
  it('matches the snapshot', () => {
    //* Act
    const { getByTestId } = render(<ProductList data-testid='testProductList' />);

    //* Assert
    const testProductList = getByTestId('testProductList');

    // Snapshot test
    expect(testProductList).toMatchSnapshot();
  });
});
