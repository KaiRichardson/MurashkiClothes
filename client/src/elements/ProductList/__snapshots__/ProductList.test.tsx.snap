// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> tests matches the snapshot 1`] = `
.c0 {
  --gap: 1.5rem;
  padding: var(--gap) 0;
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  grid-gap: var(--gap);
}

@media screen and (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media screen and (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

<section
  class="c0"
  data-testid="testProductList"
/>
`;
